main.c

```diff
@@ -3,25 +3,98 @@
-    printf("*****Searched Results*****\n");
+    Book* searched_books = searchBooks("Fantasy", library);
+    printf("*****Searched Book Results*****\n");
     printBooks(searched_books);
-    Book* filterBy = createBook(NULL, NULL, -1, NULL, -1, 10);
-    Book* filtered_books = filterBooks(filterBy, library);  // Expecting two records
-    printf("*****Filtered Results*****\n");
+    enum book_tag filter_tags[] = { fantasy, fiction};
+    Book* filterBy = createBook(NULL, NULL, 0, no_book_status,filter_tags, 2, 0, 0, 0);
+    Book* filtered_books = filterBooks(filterBy, library);
+    printf("*****Filtered Book Results*****\n");
     printBooks(filtered_books);
 
+    freeBookList(searched_books);
+    freeBookList(filtered_books);
+
+    User* new_user1 = createUser(1111, "Luke Skywalker", "luke.s", "luke", "luke.s@gmail.com", author, active_user);
+    User* new_user2 = createUser(2222, "Obiwan Kenobi", "obiwan.k", "obiwan", "obiwan.k@gmail.com", customer, active_user);
+    User* new_user3 = createUser(3333, "Darth Vader", "darth.v", "darth", "darth.v@gmail.com", librarian, active_user);
+    insertUser(new_user1, &user_list_head);
+    insertUser(new_user2, &user_list_head);
+    insertUser(new_user3, &user_list_head);
+    printf("***USERS***\n");
+    printUser(user_list_head);
+    User* searched_users = searchUsers("VADER", user_list_head);
+    printf("*****Searched User Results*****\n");
+    printUser(searched_users);
+    User* user_parameters = createUser(-1, NULL, NULL, NULL, NULL, customer, active_user);
+    User* filtered_users = filterUsers(user_parameters, user_list_head);
+    printf("*****Filtered User Results*****\n");
+    printUser(filtered_users);
+
+    freeUserList(searched_users);
+    freeUserList(filtered_users);
+
+    BookCopy* copy1 = createBookCopy(01, "1234", active);
+    BookCopy* copy2 = createBookCopy(02, "1234", active);
+    BookCopy* copy3 = createBookCopy(03, "1234", active);
+    insertBookCopy(copy1, &inventory);
+    insertBookCopy(copy2, &inventory);
+    insertBookCopy(copy3, &inventory);
+    printf("***INVENTORY***\n");
+    printBookCopies(inventory);
+
+    Date date1 = {2023, 6, 29};
+    Date date2 = {2023, 2, 06};
+    Date date3 = {2023, 8, 11};
+    BookTransaction* rental1 = createTransaction(01, &date1, 1111, open);
+    BookTransaction* rental2 = createTransaction(02, &date2, 2222, open);
+    BookTransaction* rental3 = createTransaction(03, &date3, 3333, open);
+    insertTransaction(rental1, &transaction_list);
+    rentBook(rental1, library, inventory);
+    insertTransaction(rental2, &transaction_list);
+    rentBook(rental2, library, inventory);
+    insertTransaction(rental3, &transaction_list);
+    rentBook(rental3, library, inventory);
+    printf("***TRANSACTIONS***\n");
+    printTransactions(transaction_list);
+    BookTransaction* searched_trans = searchTransaction("Luke", transaction_list, user_list_head); // use name to find user id linked w trans
```

Comment:
I think we need to change flow of building transaction.
1) Transaction struct should have a copyBook struct instance, that shows which book has been rented out by which user
2) rentBook function should not take copyBook instance as paremeter, the main purpose of rentBook function is to check is there any instance of copyBook available if yes then create a transaction instance 
3) It means creating a transaction should come inside the rentBook function

Reply:
I thought we already agreed on existing structs (Book, BookCopy, User, Transaction) before switching to CLion. Currently Transaction struct only contain book unique ID (book_uid) and from that we know which copy was rented out and not having to use extra space for the whole Bookcopy struct inside of Transaction?

Reply:
**Rent a book** 
  1) Any book copy available (book_id, book_copy_head) -> (NULL, book_uuid)
  2) Transaction 
  3) Print status of transaction